lib/sevak/publisher.rb

```diff
@@ -6,11 +6,25 @@ class Publisher
     include Singleton
 
     def self.publish(queue_name, message)
-      instance.queue(queue_name).publish(message.to_json)
+      attempt = 0
+      begin
+        instance.queue(queue_name).publish(message.to_json)
+      rescue => e #Bunny::ConnectionClosedError => e
+        attempt += 1
+        sleep(0.001)
+        attempt < 2 ? retry : raise
+      end
     end
 
     def self.delayed_publish(queue_name, message, delay = 0)
-      instance.publish_exchange(queue_name, message, delay.to_i)
+      attempt = 0
```

Comment:
Firstly these the code is duplicated here, please move the block to a common method "publish_message" and move the code there. Secondly rescue => e, catches all sort of error not just Bunny::ConnectionClosedError. But we want to rescue a single exception here, thirdly remove the unnecessary comment. Finally the raise should throw the exact error that is raised here.